File: CodeLab/EnjoyArkUIX/.arkui-x/ios/app/PlatformView/MyWebview.m
/*
 * Copyright (c) 2025 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
 
#import <Foundation/Foundation.h>
#import <WebKit/WebKit.h>
#import "MyWebview.h"

@interface MyWebview ()

@end


@implementation MyWebview {
    WKWebView *webView;
    NSString* viewTag;
}

- (instancetype)initWithFrame
{
    self = [super init];
    if (self) {
        viewTag = @"WebView";
        WKWebViewConfiguration *configuration = [[WKWebViewConfiguration alloc] init];
        webView = [[WKWebView alloc] initWithFrame:CGRectZero configuration:configuration];
        NSURL *url = [NSURL URLWithString:@"https://m.vmall.com/portal/activity/index.html?pn=YDJK&cid=178094&callapp=no"];
        NSURLRequest *request = [NSURLRequest requestWithURL:url];
        [webView loadRequest:request];
        webView.frame = CGRectMake(0, 0, 50, 50);
    }
    return self;
}

- (UIView*) view {
    return webView;
}

- (void)onDispose {
    webView = nil;
}

- (NSString*) getPlatformViewID {
    return viewTag;
}
@end

File: CodeLab/EnjoyArkUIX/.arkui-x/android/app/src/main/java/com/example/enjoyarkuix/platformview/MyWebView.java
/*
 * Copyright (c) Huawei Technologies Co., Ltd. 2025-2025. All rights reserved.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.example.enjoyarkuix.platformview;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.View;
import android.webkit.WebView;

import ohos.ace.adapter.capability.platformview.IPlatformView;

/**
 * MyWebView.
 *
 * @since 20
 */
public class MyWebView implements IPlatformView {
    private String id = "WebView";
    private WebView mWebView;

    @SuppressLint("SetJavaScriptEnabled")
    public MyWebView(Context context) {
        mWebView = new WebView(context);
        String url = "https://developer.huawei.com/";
        mWebView.getSettings().setJavaScriptEnabled(true);
        mWebView.getSettings().setDomStorageEnabled(true);
        mWebView.loadUrl(url);
    }

    @Override
    public View getView() {
        return mWebView;
    }

    @Override
    public void onDispose() {
        mWebView.destroy();
    }

    @Override
    public String getPlatformViewID() {
        return id;
    }
}
